src/double.ts

```diff
@@ -48,18 +48,24 @@ export class Double extends BSONValue {
    */
   static fromString(value: string): Double {
     const coercedValue = Number(value);
-    const nonFiniteValidInputs = ['Infinity', '-Infinity', 'NaN'];
+
+    if (value === 'NaN') {
+      return new Double(NaN);
+    } else if (value === 'Infinity') {
+      return new Double(Infinity);
+    } else if (value === '-Infinity') {
+      return new Double(-Infinity);
+    }
 
     if (value.trim() !== value) {
       throw new BSONError(`Input: '${value}' contains whitespace`);
+    } else if (!Number.isFinite(coercedValue) && !Number.isNaN(coercedValue)) {
```

Comment:
check `Infinity/-Infinity` case before `NaN` case because this allows more descriptive 'invalid characters' error a better code path